homeassistant/components/myuplink/binary_sensor.py

```diff
@@ -98,3 +136,68 @@ def is_on(self) -> bool:
         """Binary sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
         return int(device_point.value) != 0
+
+    @property
+    def available(self) -> bool:
+        """Return device data availability."""
+        return super().available and (
+            self.coordinator.data.devices[self.device_id].connectionState
+            == DeviceConnectionState.Connected
+        )
+
+
+class MyUplinkDeviceBinarySensor(MyUplinkEntity, BinarySensorEntity):
+    """Representation of a myUplink device bound binary sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        entity_description: BinarySensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the binary_sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool:
+        """Binary sensor state value."""
+        return (
+            self.coordinator.data.devices[self.device_id].connectionState
+            == DeviceConnectionState.Connected
+        )
+
+
+class MyUplinkSystemBinarySensor(MyUplinkSystemEntity, BinarySensorEntity):
+    """Representation of a myUplink system bound binary sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        system_id: str,
+        device_id: str,
+        entity_description: BinarySensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the binary_sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            system_id=system_id,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool:
+        """Binary sensor state value."""
+        return self.coordinator.data.systems[0].has_alarm
```

Comment:
If you have more than one system, will this be correct? I thought you should find the correct system based on `system_id`?

Reply:
Good catch. I will correct it.